lib/Common/CommonPal.h

```diff
@@ -4,6 +4,12 @@
 //-------------------------------------------------------------------------------------------------------
 #pragma once
 
+// Define _ALWAYSINLINE for template that want to always inline, but doesn't allow inline linkage in clang
+#if defined(__GNUC__) || defined(__clang__)
+#define _ALWAYSINLINE __attribute__((always_inline))
```

Comment:
We need inline linkage for __forceinline when they are defined in the header.
But template can't have inline linkage in clang.  That's why we need a different define to only have the always_inline attribute.